### PR TITLE
Feature/filter transaction

### PR DESCRIPTION
**Pull request by melanie-adler**

### CODE REVIEW

**Review by JessicaLoers — COMMENTED** [review]
**Well done. Keep on the good work @melanie-adler** ✨

Just a few minor suggestions, but one thing I’d really like to point out:

By calculating `filteredTransactions` directly you can learn a lot about state handling and derived state. This way, you can remove the extra `useEffect` and `filteredTransactions` state, making the component cleaner and easier to maintain. 

**Give this approach a try! 🙃**

File: components/TransactionsList.js [hunk 1]
```diff
@@ -28,12 +28,18 @@ const groupTransactionByDate = (transactions) => {
...
   handleDeleteTransaction,
   handleEditTransaction,
   handleOpenEditMode,
   openModal,
 }) {
   const groupedTransactions = groupTransactionByDate(transactions);
+  const emptyListMessage =
+    transactions.length === 0 && selectedCategory !== ""
+      ? "No matches found"
+      : "No transactions available. Please add a new transaction.";
```
[the remark this answers is not in the document]
Comment (melanie-adler) [reply components/TransactionsList.js:41]: I implemented your suggestion. Thanks!

File: pages/index.js [hunk 2]
```diff
@@ -82,11 +109,25 @@ export default function HomePage() {
...
   }
 
+  function toggleFilter() {
+    setIsFilterSelectOpen(!isFilterSelectOpen);
+  }
+
+  function filterTransactions(category) {
+    setIsFilterSelectOpen(false);
+    setSelectedCategory(category);
+  }
```
[the remark this answers is not in the document]
Comment (melanie-adler) [reply pages/index.js:119]: Thanks! Works great and the code is much clearer!

**Review by JessicaLoers — APPROVED** [review]
**Well done ✨**

**Pull request merged**